src/repo/iRepository.ts

```diff
@@ -28,73 +27,107 @@ interface IRepo<Task>{
 class taskRepository implements IRepo<Task>  {
     
     async create(T: Task){
-        const [results, metadata] = await instances.dbs.todoDB.sequelize.query(
-            'INSERT INTO tasks VALUES ( ?, ?, ?, ?, ?)', {
-                replacements: [T.id, T.createdAt, T.title, T.dueDate, T.completed]
-            } )
-        return results
+        try {
+            const taskModel = taskModelLink.taskModel();
+            const createTask = await taskModel.create({              
+                id: T.id,
+                createdAt: T.createdAt,
+                title: T.title,
+                dueDate: T.dueDate,
+                completed: T.completed
+            });
+            return createTask
+        } catch (err) {
+            return null
+        }
     }
 
-    async getAll(){
-        try{
-            const [results, metadata] = await instances.dbs.todoDB.sequelize.query('SELECT * FROM tasks')
-            return results
-        } catch(err){
+    async getAll(){        
+        try {
+            const taskModel = taskModelLink.taskModel();
+            const getAllTasks = await taskModel.findAll();
+            return getAllTasks
+        } catch (err) {
             return null
         }
     }
  
     async getId(new_id){
-        try{
-            const [results, metadata] = await instances.dbs.todoDB.sequelize.query('SELECT * FROM tasks WHERE id = ?',{
-                replacements: [new_id]
-            })
-            return results
-        } catch(err){
+        try {
+            const taskModel = taskModelLink.taskModel();
+            const getATasks = await taskModel.findAll({
+                where: {
+                    id: new_id
+                }
+            });
+            return getATasks
+        } catch (err) { 
             return null
         }
     }
 
     async getCompleted(val) {
-        try{
-            const [results, metadata] = await instances.dbs.todoDB.sequelize.query('SELECT * FROM tasks WHERE completed = ?',{
-                replacements: [val]
-            })
-            return results
-        } catch(err){
+        if (val == 0 || val == 1) {
```

Comment:
So the if statement works but the if statement is not necessary. You can get the same result by removing the if statement and improve bigO

`const result = await taskModels.findAll({
            where: {
                completed: val
            }
        })`